### ISSUE TITLE
Bug in resolveTypeReferenceDirectives using TypeScript 4.7+

### ISSUE THREAD

**Issue by elchininet** [wessberg/rollup-plugin-ts#181]
- Version: 2.0.7
- Rollup Version: 2.74.1
- Operating System and version (if applicable): MacOS 11.6.2
- Node Version (if applicable): 12.22.0 / 16.10.0
- Does it work with `tsc` (if applicable): Yes

### Reproduction

* Clone this project: https://github.com/elchininet/isometric
* Run `yarn`
* Run `yarn build` (it compiles without any issue)
* Update TypeScript version to 4.7.2
* Run `yarn build` again (the next error will be thrown)

>[!] (plugin Typescript) TypeError: path.charCodeAt is not a function

### Expected Behavior

Should not break

### Actual Behavior

The plugin is broken with this version of TypeScript

### Code Debug

At the moment, the parameter `typeReferenceDirectiveNames` of `CompilerHost.resolveTypeReferenceDirectives` [could be a `string[]` or a `FileReference[]`](https://github.com/microsoft/TypeScript/blob/1fb2b2d70fcd97c14a4fa1c74c28cf53497fc205/src/compiler/types.ts#L6788). The shape of `FileReference` is [the next one](https://github.com/microsoft/TypeScript/blob/1fb2b2d70fcd97c14a4fa1c74c28cf53497fc205/src/compiler/types.ts#L3189):

```typescript
interface FileReference {
    pos: number;
    end: number;
    fileName: string;
    resolutionMode?: SourceFile["impliedNodeFormat"];
}
```

The `CompilerHost.resolveTypeReferenceDirectives` method of the plugin [is expecting just a `string[]`](https://github.com/wessberg/rollup-plugin-ts/blob/cfe3913a43cfbb577c6d65acb4b12dbd4f801fe8/src/service/compiler-host/compiler-host.ts#L512). When a file with a triple slash reference is parsed, what is received in the `typeReferenceDirectiveNames` property of this method is something like this:

```javascript
[ { pos: 633, end: 637, fileName: 'node' } ]
```
And this object is used as the `id` for the `resolveId` method. In these cases, the compilation crashes because `TypeScript` tries to execute `charCodeAt` in an object in place of a string.

I did a small change just to test:

```typescript
resolveTypeReferenceDirectives(typeReferenceDirectiveNames, containingFile) {
    const resolvedTypeReferenceDirectives = [];
    for (const typeReferenceDirectiveName of typeReferenceDirectiveNames) {
        // try to use standard resolution
        const result = resolveId({
            moduleResolutionHost: this,
            parent: containingFile,
            id: typeof typeReferenceDirectiveName === 'object'
                ? typeReferenceDirectiveName.fileName
                : typeReferenceDirectiveName,
            resolveCache: this.options.resolveCache
        });
        ...
}
```

And the compilation finishes without any error.

**Comment by wessberg** [comment]
Hey there. Yeah, I saw this error too when I tested for TypeScript 4.7 support  Thanks for your investigation! I will fix it and whatever else might come up as part of the next minor release.

**Comment by GuillaumeRahbari** [comment]
https://github.com/wessberg/rollup-plugin-ts/issues/181 wait for this to be fixed

**Comment by wessberg** [comment]
This has been fixed as part of [v3.0.0[(https://github.com/wessberg/rollup-plugin-ts/releases/tag/v3.0.0)
